Sources/NIOTestUtils/ByteToMessageDecoderVerifier.swift

```diff
@@ -0,0 +1,211 @@
+    ///
+    ///     var exampleInput1 = channel.allocator.buffer(capacity: 16)
+    ///     exampleInput1.writeString("example-in1")
+    ///     var exampleInput2 = channel.allocator.buffer(capacity: 16)
+    ///     exampleInput2.writeString("example-in2")
+    ///     let expectedInOuts = [(exampleInput1, ExampleDecoderOutput("1")),
+    ///                           (exampleInput2, ExampleDecoderOutput("2"))
+    ///                          ]
+    ///     XCTAssertNoThrow(try ByteToMessageDecoderVerifier.verifyDecoder(inputOutputPairs: expectedInOuts,
+    ///                                                                     decoderFactory: { ExampleDecoder() }))
+    public static func verifyDecoder<Decoder: ByteToMessageDecoder>(inputOutputPairs: [(ByteBuffer, [Decoder.InboundOut])],
+                                                                    decoderFactory: @escaping () -> Decoder) throws where Decoder.InboundOut: Equatable {
+        typealias Out = Decoder.InboundOut
+
+        func verifySimple(channel: RecordingChannel) throws {
+            for (input, expectedOutputs) in inputOutputPairs.shuffled() {
+                try channel.writeInbound(input)
+                for expectedOutput in expectedOutputs {
+                    guard let actualOutput = try channel.readInbound(as: Decoder.InboundOut.self) else {
+                        throw VerificationError<Out>(inputs: channel.inboundWrites,
+                                                     errorCode: .noOutputResult(expectedOutput))
+                    }
+                    guard actualOutput == expectedOutput else {
+                        throw VerificationError<Out>(inputs: channel.inboundWrites,
+                                                     errorCode: .wrongDecodingResult(actual: actualOutput,
+                                                                                     expected: expectedOutput))
+                    }
+                }
+                let actualExtraOutput = try channel.readInbound(as: Decoder.InboundOut.self)
+                guard actualExtraOutput == nil else {
+                    throw VerificationError<Out>(inputs: channel.inboundWrites,
+                                                 errorCode: .unexpectedDecodingResult(actualExtraOutput!))
+                }
+            }
+        }
+
+        func verifyDripFeed(channel: RecordingChannel) throws {
+            for _ in 0..<10 {
+                for (input, expectedOutputs) in inputOutputPairs.shuffled() {
+                    for c in input.readableBytesView {
+                        var buffer = channel.allocator.buffer(capacity: 12)
+                        buffer.writeString("BEFORE")
+                        buffer.writeInteger(c)
+                        buffer.writeString("AFTER")
+                        buffer.moveReaderIndex(forwardBy: 6)
+                        buffer.moveWriterIndex(to: buffer.readerIndex + 1)
```

Comment:
Might it be clearer to move the `buffer` setup dance outside the loop, and then just have the loop body do `setInteger(_:at:)` instead?

Reply:
done